skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
```

Comment:
Just timed it, my calculation of the coefficients is 20x faster than the scipy version, because I can split up the calculation

```
# scipy 
>>> x = np.arange(1, 1000)
>>> y = np.arange(1, 1000)
>>> A = np.vstack([x, np.ones(len(x))]).T
>>> lstsq(A, y)[0]
array([  1.00e+00,   2.98e-14])

>>>  %timeit lstsq(A, y)
10000 loops, best of 3: 63 us per loop

# mine
>>> Y = np.atleast_2d(y)
>>> X = np.vstack([np.ones(len(x)), x]).T
>>> XX1 = np.linalg.pinv(X.T.dot(X))
>>> XX1.dot(X.T.dot(Y))
array([[ -5.93e-11],
       [  1.00e+00]])

>>> %timeit XX1.dot(X.T.dot(Y))
The slowest run took 8.61 times longer than the fastest. This could mean that an intermediate result is being cached 
100000 loops, best of 3: 3.71 us per loop
```